Auto update failing for ipfs

Hi, I'm Termux 🤖.

I'm here to help you update your Termux packages.

I've tried to update the [ipfs](https://github.com/termux/termux-packages/blob/44ecaaf21d7ebe2bea7af6b1be6b8fe353c49c0b/packages/ipfs) package, but it failed.

Here's the output of the update script:
<details>
<summary>Show log</summary>
<pre lang="console">
INFO: Updating ipfs [Current version: 0.30.0]
INFO: package being updated to 0.31.0.
Downloading https://github.com/ipfs/kubo/releases/download/v0.31.0/kubo-source.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

 58 17.2M   58 10.0M    0     0  12.1M      0  0:00:01 --:--:--  0:00:01 12.1M
100 17.2M  100 17.2M    0     0  18.7M      0 --:--:-- --:--:-- --:--:-- 75.6M

You are about to commit these changes:

--------------------
diff --git a/packages/ipfs/build.sh b/packages/ipfs/build.sh
index 4759df309..95609aabd 100644
--- a/packages/ipfs/build.sh
+++ b/packages/ipfs/build.sh
@@ -3,9 +3,9 @@ TERMUX_PKG_DESCRIPTION="A peer-to-peer hypermedia distribution protocol"
 TERMUX_PKG_LICENSE="MIT, Apache-2.0"
 TERMUX_PKG_LICENSE_FILE="LICENSE, LICENSE-APACHE, LICENSE-MIT"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="0.30.0"
+TERMUX_PKG_VERSION="0.31.0"
 TERMUX_PKG_SRCURL=https://github.com/ipfs/kubo/releases/download/v${TERMUX_PKG_VERSION}/kubo-source.tar.gz
-TERMUX_PKG_SHA256=f863b6dc0f431c81ee2b318d90f2c136fbcfcc52f715f418458bbd4df3e28b4f
+TERMUX_PKG_SHA256=6e4f44e8ae5e388a0f51115a5a7c178b2521a99c56ea4fe99a91af4e0acce108
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_SUGGESTS="termux-services"
 TERMUX_PKG_SERVICE_SCRIPT=("ipfs" "[ ! -d \"${TERMUX_ANDROID_HOME}/.ipfs\" ] && ipfs init --empty-repo 2>&1 && ipfs config --json Swarm.EnableRelayHop false 2>&1 && ipfs config --json Swarm.EnableAutoRelay true 2>&1; exec ipfs daemon --enable-namesys-pubsub 2>&1")
--------------------

bump(main/ipfs): 0.31.0


Not committing to Git!

INFO: Trying to build package.
INFO: Package ipfs exists in main repo.
Running container 'termux-package-builder' from image 'ghcr.io/termux/package-builder'...
termux - building ipfs for arch aarch64...
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13134  100 13134    0     0  22598      0 --:--:-- --:--:-- --:--:-- 22605
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   4039      0 --:--:-- --:--:-- --:--:--  4043
gpg: Signature made Wed 16 Oct 2024 08:21:28 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13320  100 13320    0     0  23962      0 --:--:-- --:--:-- --:--:-- 23956
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   1349      0 --:--:-- --:--:-- --:--:--  1347
100   833  100   833    0     0   1349      0 --:--:-- --:--:-- --:--:--  1347
gpg: Signature made Mon 14 Oct 2024 11:50:11 AM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13120  100 13120    0     0  23141      0 --:--:-- --:--:-- --:--:-- 23180
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   1523      0 --:--:-- --:--:-- --:--:--  1522
100   833  100   833    0     0   1522      0 --:--:-- --:--:-- --:--:--  1522
gpg: Signature made Wed 16 Oct 2024 04:25:05 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://github.com/ipfs/kubo/releases/download/v0.31.0/kubo-source.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

  0 17.2M    0 39649    0     0   101k      0  0:02:55 --:--:--  0:02:55  101k
100 17.2M  100 17.2M    0     0  39.5M      0 --:--:-- --:--:-- --:--:--  325M
Applying patch: go-ipfs-0.4.15_bin_dist_get.patch
Downloading https://golang.org/dl/go1.22.6.linux-amd64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100    80  100    80    0     0    727      0 --:--:-- --:--:-- --:--:--   733

100    75  100    75    0     0    353      0 --:--:-- --:--:-- --:--:--   353

100 65.7M  100 65.7M    0     0  93.2M      0 --:--:-- --:--:-- --:--:-- 93.2M
go version go1.23.0 linux/amd64
bin/check_go_version 1.18
plugin/loader/preload.sh > plugin/loader/preload.go
go fmt plugin/loader/preload.go >/dev/null
go build "-trimpath" -ldflags="-X "github.com/ipfs/kubo".CurrentCommit=5a32936f7" -o "cmd/ipfs/ipfs" "github.com/ipfs/kubo/cmd/ipfs"
# github.com/ipfs/kubo/cmd/ipfs
link: github.com/wlynxg/anet: invalid reference to net.zoneCache
make: *** [cmd/ipfs/Rules.mk:22: cmd/ipfs/ipfs] Error 1
ERROR: failed to build.</pre>
</details>

<hr>
<i>
Above error occured when I last tried to update at 2024-10-17 02:06:25 UTC.<br>
Run ID: <a href="https://github.com/termux/termux-packages/actions/runs/11375976027">11375976027</a><br><br>
<b>Note:</b> Automatic updates will be disabled until this issue is resolved.<br>
</i>

--- a/packages/ipfs/build.sh
+++ b/packages/ipfs/build.sh
@@ -3,14 +3,12 @@ TERMUX_PKG_DESCRIPTION="A peer-to-peer hypermedia distribution protocol"
 TERMUX_PKG_LICENSE="MIT, Apache-2.0"
 TERMUX_PKG_LICENSE_FILE="LICENSE, LICENSE-APACHE, LICENSE-MIT"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="0.30.0"
+TERMUX_PKG_VERSION="0.31.0"
 TERMUX_PKG_SRCURL=https://github.com/ipfs/kubo/releases/download/v${TERMUX_PKG_VERSION}/kubo-source.tar.gz
-TERMUX_PKG_SHA256=f863b6dc0f431c81ee2b318d90f2c136fbcfcc52f715f418458bbd4df3e28b4f
+TERMUX_PKG_SHA256=6e4f44e8ae5e388a0f51115a5a7c178b2521a99c56ea4fe99a91af4e0acce108
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_SUGGESTS="termux-services"
 TERMUX_PKG_SERVICE_SCRIPT=("ipfs" "[ ! -d \"${TERMUX_ANDROID_HOME}/.ipfs\" ] && ipfs init --empty-repo 2>&1 && ipfs config --json Swarm.EnableRelayHop false 2>&1 && ipfs config --json Swarm.EnableAutoRelay true 2>&1; exec ipfs daemon --enable-namesys-pubsub 2>&1")
-# See https://github.com/ipfs/kubo/issues/10501 about issues with ipfs and go 1.23:
-TERMUX_PKG_GO_USE_OLDER=true
 
 termux_step_make() {
 	termux_setup_golang

--- a/packages/ipfs/cmd-ipfs-Rules.mk.patch
+++ b/packages/ipfs/cmd-ipfs-Rules.mk.patch
@@ -1,0 +1,14 @@
+See https://github.com/wlynxg/anet?tab=readme-ov-file#how-to-build-with-go-1230-or-later
+
+diff -u -r ../kk/cmd/ipfs/Rules.mk ./cmd/ipfs/Rules.mk
+--- ../kk/cmd/ipfs/Rules.mk	2024-10-16 18:28:08.000000000 +0000
++++ ./cmd/ipfs/Rules.mk	2024-10-17 21:18:34.984576632 +0000
+@@ -13,7 +13,7 @@
+ # DEPS_OO_$(d) += merkledag/pb/merkledag.pb.go namesys/pb/namesys.pb.go
+ # DEPS_OO_$(d) += pin/internal/pb/header.pb.go unixfs/pb/unixfs.pb.go
+ 
+-$(d)_flags =-ldflags="-X "github.com/ipfs/kubo".CurrentCommit=$(git-hash)"
++$(d)_flags =-ldflags="-checklinkname=0 -X "github.com/ipfs/kubo".CurrentCommit=$(git-hash)"
+ 
+ $(d)-try-build $(IPFS_BIN_$(d)): GOFLAGS += $(cmd/ipfs_flags)
+ 

